manifests/infra/linux.pp

```diff
@@ -0,0 +1,58 @@
+# Class: newrelic::infra::linux
+# =============================
+#
+# This class install and start the New Relic Infrastrucure Agent which replaces the deprecated server class
+#
+# Parameters
+# ----------
+#
+# * `newrelic_license_key`
+#   The accounts license key as issued by New Relic
+# * `newrelic_manage_repo`
+#   Boolean flag to configure the New Relic repo
+#
+class newrelic::infra::linux (
+  $newrelic_license_key = undef,
+  $newrelic_manage_repo = true,
+) {
+
+  if ! $newrelic_license_key {
+    fail('You must specify a valid License Key.')
+  }
+
+  if $newrelic_manage_repo {
+
+    case $facts['os']['family'] {
+      'redhat': {
+        file { '/etc/yum.repos.d/newrelic-infra.repo':
```

Comment:
Any reason why we don't use a Puppet `yumrepo` resource here?

https://docs.puppet.com/puppet/latest/types/yumrepo.html